packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketVerify/CoinmarketVerify.tsx

```diff
@@ -127,6 +132,15 @@ export const CoinmarketVerify = ({ coinmarketVerifyAccount, currency }: Coinmark
         },
     });
 
+    // close modals and reset addressVerified on device connection change
+    useEffect(() => {
+        dispatch({
+            type: COINMARKET_BUY.VERIFY_ADDRESS,
+            addressVerified: undefined,
+        });
+        dispatch(modalActions.onCancel());
+    }, [device?.connected, dispatch]);
+
```

Comment:
Perhaps this should be moved inside the modal. It is only related to `ProceedUnverifiedAddressModal`, isn't it?

If you use `ConfirmUnverifiedModal`, the `useEffect` is already there.

Reply:
Yes, it is related. But it is not possible to clear the  `addressVerified`, if the modal is not opened.

Reply:
It is only set in the modal. But It is also necessary to set the value as `undefined` if the modal is not active and the connection of Trezor will change.